File: skillful_network_server/src/main/java/fr/uca/cdr/skillful_network/entities/application/JobApplication.java
package fr.uca.cdr.skillful_network.entities.application;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import fr.uca.cdr.skillful_network.entities.user.User;

import javax.persistence.*;
import java.util.Date;

@Entity
public class JobApplication extends Application {

    @ManyToOne(cascade = CascadeType.REFRESH)
//    @JoinColumn(name = "job_offer_id")
    @JsonManagedReference
    private JobOffer offer;

    public JobApplication() {

    }

    public JobApplication(User user, JobOffer offer) {
        this.user = user;
        this.offer = offer;
    }

    public JobApplication(User user, ApplicationStatus status, Date submitDate, JobOffer offer) {
        super(user, status, submitDate);
        this.offer = offer;
    }

    public JobOffer getOffer() {
        return offer;
    }

    public void setOffer(JobOffer offer) {
        this.offer = offer;
    }
}

File: skillful_network_client/src/app/shared/models/application/job-application.ts
import { Application } from './application';
import { JobOffer } from './job-offer';

export class JobApplication extends Application {

    private _jobOffer: JobOffer;

    constructor(data: any) {
       super(data);
       this._jobOffer = data.offer;
    }

    public set jobOffer(jobOffer: JobOffer) {
        this._jobOffer = jobOffer;
    }

    public get jobOffer() : JobOffer {
        return this._jobOffer;
    }

}

